lib/nokogiri/xml/parse_options.rb

```diff
@@ -1,167 +1,406 @@
+    #
+    # ```
+    # XML::parse(xml_s) {|options| puts options.inspect }
+    # #<Nokogiri::XML::ParseOptions: @options=4196353 recover, nonet, big_lines, default_xml, default_schema>
+    # HTML4::parse(html_s) {|options| puts options.inspect }
+    # #<Nokogiri::XML::ParseOptions: @options=4196449 recover, nowarning, nonet, big_lines, default_html, default_xml, noerror, default_schema>
+    # ```
+    #
+    # When the block returns, the parsing is performed using those `options`.
+    #
+    # The block may modify those options, which affects parsing:
+    #
+    # ```
+    # bad_xml = '<root>'                              # End tag missing.
+    # XML::parse(bad_xml)                             # No error because option RECOVER is on.
+    # XML::parse(bad_xml) {|options| options.strict } # Raises SyntaxError because option STRICT is on.
+    # ```
+    #
+    # ## Convenience Methods
+    #
+    # A \ParseOptions object has three sets of convenience methods,
+    # each based on the name of one of the constants:
+    #
+    # - **Setters**: each is the downcase of an option name, and turns **on** an option:
+    #
+    #     ```
+    #     options = ParseOptions.new
+    #     # => #<Nokogiri::XML::ParseOptions: ... strict>
+    #     options.big_lines
+    #     # => #<Nokogiri::XML::ParseOptions: ... strict, big_lines>
+    #     options.compact
+    #     # => #<Nokogiri::XML::ParseOptions: ... strict, compact, big_lines>
+    #     ```
+    #
+    # - **Unsetters**: each begins with `no`, and turns **off** an option.
+    #   Note that there is no unsetter `nostrict`,
+    #   but the setter `recover` serves the same purpose:
+    #
+    #     ```
+    #     options.nobig_lines
+    #     # => #<Nokogiri::XML::ParseOptions: ... strict, compact>
+    #     options.nocompact
+    #     # => #<Nokogiri::XML::ParseOptions: ... strict>
+    #     options.recover # Functionally equivalent to nostrict.
+    #     # => #<Nokogiri::XML::ParseOptions: ... recover>
+    #     options.noent   # Set NOENT.
+    #     # => #<Nokogiri::XML::ParseOptions: ... recover, noent>
+    #     options.nonoent # Unset NOENT.
+    #     # => #<Nokogiri::XML::ParseOptions: ... recover>
+    #     ```
+    #
+    # - **Queries**: each ends with `?`, and returns whether an option is **on** or **off**:
+    #
+    #     ```
+    #     options.recover? # => true
+    #     options.strict?  # => false
+    #     ```
+    #
+    # Each setter and unsetter method returns `self`,
+    # so the methods may be chained:
+    #
+    # ```
+    # options.compact.big_lines
+    # # => #<Nokogiri::XML::ParseOptions: ... strict, compact, big_lines>
+    # ```
     #
     class ParseOptions
-      # Strict parsing
+
+      # Strict parsing; do not recover from errors in input.
       STRICT      = 0
 
-      # Recover from errors. On by default for XML::Document, XML::DocumentFragment,
-      # HTML4::Document, HTML4::DocumentFragment, XSLT::Stylesheet, and XML::Schema.
```

Comment:
I think we need to keep the note on each constant that states whether the bit is ON or OFF by default (and which macros have which value if it varies).